### PR TITLE
Sync up CCSE and master

### PR DESCRIPTION
**Pull request by atmyers**
The "ccse" branch is starting to diverge a bit from master, probably a good idea to sync them up. 
Ideally we wouldn't use a different branch at all. 

Mostly these changes were added to support additional options that the combution folks and WarpX people wanted.

### CODE REVIEW

File: regtest.py [hunk 1]
```diff
@@ -28,39 +28,48 @@
 import test_report as report
 import test_coverage as coverage
 
+safe_flags = ['USE_CUDA', 'USE_ACC', 'USE_MPI', 'USE_OMP', 'DEBUG', 'USE_GPU']
+
```
Comment (zingale) [inline regtest.py:32]: can you add `TEST`?

**Comment by zingale** [comment]
we have additional build commands in Castro -- how does _check_safety deal with those (e.g., USE_TRUE_SDC, USE_SIMPLIFIED_SDC, NETWORK_DIR, EOS_DIR, ....)

File: regtest.py (hunk 1 above)
Comment (atmyers) [reply regtest.py:32]: sure, I added it.

**Comment by atmyers** [comment]
if there are any compile strings that are not found in that list, then it fall back to realclean. It's only in the case that we're only using compile_strings that go in the executable name that this will alter the current behavior. 

USE_GPU is a special case that is equivalent to USE_CUDA=TRUE and USE_OMP=FALSE

We added this because WarpX was spending a lot of time recompiling when it didn't strictly speaking have to.

**Pull request merged**